### PR TITLE
[TEST] Refine test suites

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Rename `KyuubiOperationHiveSuite` to `KyuubiOperationHiveCatalogSuite`;
Simplify redundant case in `KyuubiOperationPerConnectionSuite`;

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala [hunk 1]
```diff
@@ -26,7 +26,7 @@ import org.apache.kyuubi.config.KyuubiConf
 /**
  * UT with Connection level engine shared cost much time, only run basic jdbc tests.
  */
-class KyuubiOperationPerConnectionSuite extends WithKyuubiServer with BasicJDBCTests {
+class KyuubiOperationPerConnectionSuite extends WithKyuubiServer with JDBCTestUtils {
```
Comment (ulysses-you) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala:29]: We don't need to test basic tests for connection share level ?

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala:29]: The cases in `JDBCTestUtils` got tested on Engine side, and several times on Kyuubi Server side, it's quite redundant. Thus just keep connection share level specific cases in `KyuubiOperationPerConnectionSuite` to simplify the test suites.

**Comment by ulysses-you** [comment]
thanks, merging to master for v1.3.0

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala (hunk 1 above)
Comment (ulysses-you) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala:29]: make sense

**Pull request closed without merging**